plugins/PostProcessingPlugin/scripts/AddCoolingProfile.py

```diff
@@ -0,0 +1,803 @@
+                    "type": "bool",
+                    "default_value": false,
+                    "enabled": true
+                },
+                "fan_raft_percent":
+                {
+                    "label": "Raft Fan %:",
+                    "description": "Enter the percentage for the Raft.",
+                    "type": "int",
+                    "default_value": 35,
+                    "minimum_value": 0,
+                    "maximum_value": 100,
+                    "unit": "%    ",
+                    "enabled": "fan_enable_raft"
+                }
+            }
+        }"""
+
+    def execute(self, data):
+#Initialize variables that are buried in if statements.
+        T0_fan = " P0"; T1_fan = " P0"; T2_fan = " P0"; T3_fan = " P0"; is_multi_extr_print = True
+#Get some information from Cura.
+        extrud = Application.getInstance().getGlobalContainerStack().extruderList
+        fan_mode = True #This will be true when fan scale is 0-255pwm and false when it's RepRap 0-1 (Cura 5.x)
+        try: #For 4.x versions that don't have this setting
+            fan_mode = not bool(extrud[0].getProperty("machine_scale_fan_speed_zero_to_one", "value"))
+        except:
+            all
+        bed_adhesion = (extrud[0].getProperty("adhesion_type", "value"))
+        extruder_count = Application.getInstance().getGlobalContainerStack().getProperty("machine_extruder_count", "value")
+        Logger.log("d", "Add Cooling Profile - fan_mode; " + str(fan_mode) + " ; bed_adhesion; " + str(bed_adhesion) + " ; extruder_count; " + str(extruder_count))
+#Assign the fan numbers to the tools.
+        if extruder_count == 1:
+            is_multi_fan = False
+            is_multi_extr_print = False
+            if int((extrud[0].getProperty("machine_extruder_cooling_fan_number", "value"))) > 0:
+                T0_fan = " P" + str((extrud[0].getProperty("machine_extruder_cooling_fan_number", "value")))
+            else:
+                T0_fan = ""  #No P parameter if there is a single fan circuit.
+        elif extruder_count > 1:
+            is_multi_fan = True
+            T0_fan = " P" + str((extrud[0].getProperty("machine_extruder_cooling_fan_number", "value")))  
+        if is_multi_fan:
+            if extruder_count > 1: T1_fan = " P" + str((extrud[1].getProperty("machine_extruder_cooling_fan_number", "value")))
+            if extruder_count > 2: T2_fan = " P" + str((extrud[2].getProperty("machine_extruder_cooling_fan_number", "value")))
+            if extruder_count > 3: T3_fan = " P" + str((extrud[3].getProperty("machine_extruder_cooling_fan_number", "value")))
+        Logger.log("d", "Add Cooling Profile - T0 fan; " + str(T0_fan) + " ; T1_fan; " + str(T1_fan) + " ; T2 fan; " + str(T2_fan) + " ; T3 fan; " + str(T3_fan))
+#Initialize the fan_list with defaults
+        fan_list = []
+        for q in range(0,8,1):
+            fan_list.append("999999999")
```

Comment:
The "999999999" is just to insure the number is beyond what the actual top layer number might be.    The firmware won't see it.  It could also be "str(Len[data]+1)" which is 15 characters as opposed to 9.  Functionally, there really isn't any difference.